content/en/docs/guides/wait-for-ready.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Wait For Ready
+description: >-
+  Explains how and why to configure the wait for ready feature.
+---
+
+### Overview
+
+This is a feature which can be used on a stub which will cause the rpcs to wait
+(until an optional deadline is exceeded) for the server to become available
+before sending the request.
+
+This allows for robust batch workflows as transient server problems (including
+startup) won't cause job failures.
+
+When an RPC is created while the channel is not in a READY state, without
+waitForReady it will immediately return a failure; with waitForReady it will
+simply block until the connection becomes ready.
+
+For detailed semantics see: [grpc/doc/wait-for-ready.md][grpc doc]
+
+### How to use Wait for Ready
+
+waitForReady is an attribute associated with a stub. When an RPC is created, the
+wait for ready attribute is automatically provided to the associated stream.
+
+Warning: the RPC can still fail for other reasons besides the server not being
+ready, so you must still implement error handling.
+
+```mermaid
+flowchart TD
+  F(Failure Status)
+  SENT(RPC Sent to server)
+  CH(Channel) -->|Create stub &\n set waitForReady| S(Stub)
+  S --> |create RPC| RPC
+  RPC -->|Initiates Communication| CH
+  CH --> ST{Channel State}
+  ST --> | READY| SENT
+  ST -->|TRANSIENT_FAILURE / IDLE| BUR(Block on channel becoming READY or a permanent failure)
+  BUR --> |Channel becomes READY| SENT
+  BUR --> |Permanent Failure / \nSHUTDOWN| F
+  ST --> |Permanent Failure / \nSHUTDOWN| F
+```
+
+### Alternatives
+
+- Loop until RPC stops returning transient failures.
+- Implement an onReady Handler and handle your own blocking _(for languages that
```

Comment:
This isn't really an alternative.

Reply:
Good point. I changed it to say that using an onReady handler could be an optimization.   Added accepting the failures that might have been avoided in order to fail fast.